### ISSUE TITLE
MariaDB: "Keywords should not be used as identifiers" for `ALTER TABLE foo ADD SYSTEM VERSIONING;`

### ISSUE THREAD

**Issue by tgross35** [sqlfluff/sqlfluff#7879]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

The following is valid for MariaDB but raises errors in SQLFluff:

```sql
ALTER TABLE foo ADD SYSTEM VERSIONING;
```

### Expected Behaviour

No errors, the syntax is valid.

### Observed Behaviour

The following is raised:

```text
CP02 	1 / 21 	Unquoted identifiers must be consistently lower case.
RF04 	1 / 21 	Keywords should not be used as identifiers.
```

It seems like SQLFluff is expecting this to be a column addition.

### How to reproduce

https://online.sqlfluff.com/fluffed?sql=H4sIALMAEWoC_3P0CXENUghxdPJxVUjLz1dwdHFRCI4MDnH1VQhzDQr29Pfz9HO35uUCAH_IMUwoAAAA&dialect=MariaDB

### Dialect

MariaDB

### Version

`sqlfluff, version 4.2.1`

### Configuration

```text
[sqlfluff]
dialect = mariadb
```

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
